decidim-meetings/app/models/decidim/meetings/agenda_item.rb

```diff
@@ -11,7 +11,7 @@ class AgendaItem < Meetings::ApplicationRecord
 
       translatable_fields :title, :description
 
-      belongs_to :agenda, -> { order(:position) }, foreign_key: "decidim_agenda_id", class_name: "Decidim::Meetings::Agenda"
+      belongs_to :agenda, -> { joins(:agenda_items).order(:position) }, foreign_key: "decidim_agenda_id", class_name: "Decidim::Meetings::Agenda"
```

Comment:
Why do we need the order statement within the `belongs_to` relation at all? Couldn't we remove it completely?

The current version generates the following query:
```sql
SELECT "decidim_meetings_agendas".* FROM "decidim_meetings_agendas"
  WHERE "decidim_meetings_agendas"."id" = 123 ORDER BY "position" ASC LIMIT 1;
```

Which is incorrect of course but when the change is applied, the query is as follows:
```sql
SELECT "decidim_meetings_agendas".* FROM "decidim_meetings_agendas"
  INNER JOIN "decidim_meetings_agenda_items" ON "decidim_meetings_agenda_items"."decidim_agenda_id" = "decidim_meetings_agendas"."id"
  WHERE "decidim_meetings_agendas"."id" = 123 ORDER BY "position" ASC LIMIT 1;
```

It can only match one agenda as the query is filtered by its ID.

So I don't see any difference with:
```sql
 SELECT "decidim_meetings_agendas".* FROM "decidim_meetings_agendas"
  WHERE "decidim_meetings_agendas"."id" = 123 LIMIT 1;
```

There may be something here that I'm not perfectly understanding, though.

Reply:
You are right, since the query selects an single item the ORDER BY is not necessary.  I added the change because I wasn't entirely sure if any query involving this association could use that ORDER BY either.

But it also seems to me that it can be removed. If you think so, I'll simplify the PR and eliminate the ordering

Reply:
@entantoencuanto Did you notice my comment here? I'd like to get this merged before we do the 0.27 RC1 release.